Python/Product/Debugger/DebugAdapterProcess.cs

```diff
@@ -101,8 +109,22 @@ public sealed class DebugAdapterProcess : ITargetHostProcess {
             _process.ErrorDataReceived += OnErrorDataReceived;
             _process.Start();
             _process.BeginErrorReadLine();
-            if (!_connectedEvent.WaitOne(_debuggerConnectionTimeout)){
-                Debug.WriteLine("Timed out waiting for debuggee to connect.", nameof(DebugAdapterProcess));
+
+            try {
+                if (connection.Wait(_debuggerConnectionTimeout)) {
+                    var socket = connection.Result;
+                    if (socket != null) {
+                        _stream = new NetworkStream(connection.Result, ownsSocket: true);
```

Comment:
Is it possible that `_process` has exited before `_stream` is set?

Reply:
Yes, and there are a few other (unlikely) race conditions here. Karthik was going to significantly refactor this class for the next release and switch to our more reliably ProcessOutput class.

Reply:
Ok